drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -791,6 +826,200 @@ static int i2c_stm32_msg_read(const struct device *dev, struct i2c_msg *msg,
 	return -EIO;
 }
 
+static int stm32_i2c_irq_xfer(const struct device *dev, struct i2c_msg *msg,
+			      uint8_t *next_msg_flags, uint16_t slave)
+{
+	const struct i2c_stm32_config *cfg = dev->config;
+	struct i2c_stm32_data *data = dev->data;
+	I2C_TypeDef *regs = cfg->i2c;
+	bool is_timeout = false;
+
+	data->current.len = msg->len;
+	data->current.buf = msg->buf;
+	data->current.is_arlo = 0U;
+	data->current.is_nack = 0U;
+	data->current.is_err = 0U;
+	data->current.msg = msg;
+
+#if defined(CONFIG_I2C_TARGET)
+	data->master_active = true;
+#endif
+
+#if defined(CONFIG_DCACHE) && defined(CONFIG_I2C_STM32_V2_DMA)
+	if (!buf_in_nocache((uintptr_t)msg->buf, msg->len) &&
```

Comment:
Could you rather use `stm32_buf_in_nocache()` SoC helper function (here and below in this same function) in commit *"drivers: i2c_stm32_v2: Rework interrupt handler"*? You can remove implementation of local helper function `buf_in_nocache()` from commit *"drivers: i2c_stm32_v2: Remove obsolete functions"* that is anyway removed in the last commit of this series.